src/navigation/_ComponentToolbar.jsx

```diff
@@ -36,10 +31,17 @@ const defaultProps = {
 const ComponentToolbar = ({
   onToggle,
   menuIsVisible,
-  devToolsConfig,
 }) => {
-  const hasThemes = devToolsConfig && devToolsConfig.themes && devToolsConfig.themes.length > 1;
-  const hasLocales = devToolsConfig && devToolsConfig.locales && devToolsConfig.locales.length > 1;
+  const appSettings = React.useContext(AppSettingsContext);
+  const hasThemes = appSettings.themes && Object.keys(appSettings.themes).length > 1;
+  const hasLocales = appSettings.locales && appSettings.locales.length > 1;
+
+  const onChangeTheme = (theme) => (
+    appSettings.onUpdate({ ...appSettings.state, theme })
```

Comment:
don't we only want to update the theme value? I would expect the onUpdate to mantain the previous theme locale and direction states.

Reply:
yes